pkg/stats/index.go

```diff
@@ -19,24 +119,111 @@ var indexTmpl = template.Must(template.New("index").Parse(`<!DOCTYPE html><html>
 </style>
 </head>
 <body>
-Things to do:
-{{range .}}
+Things to do now:
+{{range .Profiles}}
 <h2><a href={{.Href}}>{{.Name}}</a></h2>
 <p>
 {{.Desc}}
 </p>
 {{end}}
+<br>
+<br>
+<h2>Snapshot Format & Filters</h2>
+<p>All options apply to both snapshots. Filters are inclusive.</p>
+<br>
+<label style="vertical-align:top" for="format">Choose a format:</label>
+<select style="vertical-align:top" name="format" id="format" onchange="changeFormat()">
+{{range .Formats}}
+  <option value="{{.}}">{{.}}</option>
+{{end}}
+</select>
+<label style="vertical-align:top" for="cluster">Choose cluster(s):</label>
+<select style="height: 250px; vertical-align:top" for="format" name="cluster" id="cluster" multiple="multiple" onchange="changeCluster()">
+{{range .Clusters}}
+  <option value="{{.}}">{{.}}</option>
+{{end}}
+</select>
+<label style="vertical-align:top" for="namespace">Choose namspace(s):</label>
+<select style="height: 250px; vertical-align:top" for="format" name="namespace" id="namespace" multiple="multiple" onchange="changeNamespace()">
+{{range .Namespaces}}
+  <option value="{{.}}">{{.}}</option>
+{{end}}
+</select>
+<label style="vertical-align:top" for="resourceType">Choose resource type(s):</label>
+<select style="height: 250px; vertical-align:top" for="format" name="resourceType" id="resourceType" multiple="multiple" onchange="changeResourceType()">
+{{range .ResourceTypes}}
+  <option value="{{.}}">{{.}}</option>
+{{end}}
+</select>
+<h2>Input</h2>
+<a href="/snapshots/input" id="input_url">View</a>
+<a href="/snapshots/input" id="input_url_download" download="input.json">Download</a>
+<h2>Output</h2>
+<a href="/snapshots/output" id="output_url">View</a>
+<a href="/snapshots/output" id="output_url_download" download="output.json">Download</a>
 </body>
 </html>
 `))
 
-func Index(w http.ResponseWriter, r *http.Request) {
+func (index Index) Generate(w http.ResponseWriter, r *http.Request) {
 	type profile struct {
 		Name string
 		Href string
 		Desc string
 	}
+	type pageData struct {
+		Profiles      []profile
+		Formats       []string
+		ResourceTypes []string
+		Clusters      []string
+		Namespaces    []string
+	}
 	var profiles []profile
+	clusters := make(map[string]bool)
+	namespaces := make(map[string]bool)
+
+	// Capture clusters, namespaces, resourceTypes from input.
+	// We can use these as filters for both output & input
+	input, err := index.history.GetInputCopy()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	translator := input["translator"].(map[string]interface{})
+	resourceTypes := make([]string, 0, len(translator))
+	for id := range translator {
+		resourceTypes = append(resourceTypes, id)
+	}
+
+	for resourceType, resources := range translator {
+		for _, resource := range resources.([]interface{}) {
+			resource := resource.(map[string]interface{})
+			metadata := resource["metadata"].(map[string]interface{})
+			cluster, ok := metadata["clusterName"].(string)
```

Comment:
Is this populated for input data? I had a feeling we only populated this for the output snapshot. K8s has announced they'll be deprecated the clusterName field soon, so it's not widely used atm, even though we use it in our output snapshots.

Reply:
It is yes. I am using it to get the list of clusters I can filter on and then I am using it to do the actual filtering. If this stops existing. Nothing will break. You just won't get cluster as an option when filtering